things/service.go

```diff
@@ -386,16 +389,38 @@ func (ts *thingsService) CreateChannels(ctx context.Context, token string, chann
 		return []Channel{}, errors.Wrap(ErrUnauthorizedAccess, err)
 	}
 
-	for i := range channels {
-		channels[i].ID, err = ts.idProvider.ID()
+	chs := []Channel{}
+	for _, channel := range channels {
+		ch, err := ts.createChannel(ctx, &channel, res)
 		if err != nil {
-			return []Channel{}, errors.Wrap(ErrCreateUUID, err)
+			return []Channel{}, err
 		}
+		chs = append(chs, ch)
+	}
+	return chs, nil
+}
+
+func (ts *thingsService) createChannel(ctx context.Context, channel *Channel, identity *mainflux.UserIdentity) (Channel, error) {
+	chID, err := ts.idProvider.ID()
+	if err != nil {
+		return Channel{}, errors.Wrap(ErrCreateUUID, err)
+	}
+	channel.ID = chID
+	channel.Owner = identity.GetEmail()
```

Comment:
a question, if would be switching, at some point, to identity ID instead of the email would that be a problem for policies?

Reply:
Actually, policies use ID instead of email. Here, `channel.Owner = identity.GetEmail()` email is used to store it on the database, not for policies.